src/utilities/urlSearchParams.ts

```diff
@@ -10,3 +13,9 @@ export function combineUrlSearchParams(aParams: URLSearchParams | QuerySource, b
 
   return combinedParams
 }
+
+export function appendQuery(url: Url, query: QuerySource): Url {
+  const { searchParams, ...parts } = parseUrl(url)
+
+  return createUrl({ ...parts, searchParams: combineUrlSearchParams(searchParams, query) })
+}
```

Comment:
when we're using `router.push` (or replace) and given a `url` we no longer call `resolve`. The only thing we really needed to do was combine the query that might have been provided in the resolve options arg. Now we use this function instead.

This is also used inside of `resolve` when given a non-external `url`. That way the `href` property of the ResolvedRoute includes the query from the options.